Pull Request Diplom_1

mober2000 opened the pull request.

mober2000 committed:
develop branch

mober2000 committed:
develop branch

mober2000 committed:
Merge remote-tracking branch 'origin/develop1' into develop1

mober2000 committed:
проверяю что гит нормально работает

mober2000 committed:
проверяю что гит нормально работает

mober2000 committed:
Добавил тесты на класс Burger, но пока не разобрался с dataBase, параметризацией и моками

mober2000 committed:
Диплом 1 Первая попытка

mober2000 committed:
Диплом 1 Первая попытка

pomegranate282922 commented on src/test/java/praktikum/BurgerTest.java:
```diff
@@ -0,0 +1,73 @@
+        burger.setBuns(new Bun("Булочка с корицей", 100));
+        String actual = burger.getReceipt();
+        assertEquals(reference, actual);
+    }
```
Здесь хотелось бы увидеть использование моков. Можно улучшить.

pomegranate282922 reviewed (commented):
Все хорошо, только прошу добавить ещё требуемые в задании тесты на другие классы.

mober2000 committed:
Диплом 1 Вторая попытка
Добавил тесты на классы bun и ingredient
Добавил моки в тест с выставлением булки в классе BurgerTest

pomegranate282922 commented on src/test/java/praktikum/BunTest.java:
```diff
@@ -0,0 +1,25 @@
+    public void setBunPriceTest() {
+        float expected = price;
+        float actual = bun.getPrice();
+        assertEquals(expected, actual, 0.00001);
```
Дельту лучше вынести как константу и переиспользовать.

pomegranate282922 commented on src/test/java/praktikum/BurgerTest.java:
```diff
@@ -53,11 +65,14 @@ public void setBunsTest() {
+        burger.addIngredient(ingredient);
+        float actual = burger.getPrice();
+        float expected = 30f;
+        assertEquals(expected, actual, 0.00001);
```
Отлично! Уместное использование моков.

pomegranate282922 reviewed (approved):
Зачёт!
